controllers/comment.server.controller.js

```diff
@@ -0,0 +1,54 @@
+const Comment = require('../models/comment.server.model');
+
+/**
+ * Creates a new comment for a forum post using HTTP request object data
+ * @param req HTTP request object
+ * @param res HTTP request response object
+ */
+exports.commentCreate = function (req, res) {
+    const reqBody = req.body;
+    let
+        isBadRequest = false,
+        commentParams;
+
+    // Check that every expected forum comment attribute exists in the request body
+    if (!reqBody.postID || !reqBody.authorID || !reqBody.authorDisplayName || !reqBody.bodytext || 
+        !reqBody.date || reqBody.upVotes !== 0 || reqBody.downVotes !== 0) {
+        isBadRequest = true;
+    }
+
+    if (!isBadRequest) {
+        commentParams = {
+            'postID': reqBody.postID,
+            'authorID': reqBody.authorID.length < 3 ? false : reqBody.authorID,
+            'authorDisplayName': reqBody.authorDisplayName.length < 3 ? false : reqBody.authorDisplayName,
+            'bodytext': reqBody.bodytext.length < 1 ? false : reqBody.bodytext,
+            'date': reqBody.date.length !== 10 ? false : reqBody.date,
```

Comment:
I think you should make another method that will properly check that the date is given in the correct format.